src/contracts/yas_pool.cairo

```diff
@@ -84,25 +185,304 @@ mod YASPool {
+                    state.protocol_fee += delta.try_into().unwrap();
+                };
+
+                // update global fee tracker
+                if state.liquidity > 0 {
+                    state
+                        .fee_growth_global_X128 +=
+                            FullMath::mul_div(step_fee_amount, Q128, step_fee_amount);
+                };
+
+                // shift tick if we reached the next price
+                if state.sqrt_price_X96 == step_sqrt_price_next_X96 {
+                    // if the tick is initialized, run the tick transition
+                    if step_initialized {
+                        // crosses an initialized tick
+                        let mut liquidity_net: i128 = Tick::TickImpl::cross(
+                            ref state_tick,
+                            step_tick_next,
+                            if zero_for_one {
+                                state.fee_growth_global_X128
+                            } else {
+                                self.fee_growth_global_0X128.read()
+                            },
+                            if zero_for_one {
+                                self.fee_growth_global_1X128.read()
+                            } else {
+                                state.fee_growth_global_X128
+                            },
+                            cache.seconds_per_liquidity_cumulative_X128, // TODO:
+                            cache.tick_cumulative,
+                            cache.block_timestamp
+                        );
+
+                        // if we're moving leftward, we interpret liquidityNet as the opposite sign
+                        // safe because liquidityNet cannot be type(int128).min
+                        if zero_for_one {
+                            liquidity_net = -liquidity_net;
+                        };
+
+                        state.liquidity = LiquidityMath::add_delta(state.liquidity, liquidity_net);
+                    };
+
+                    state
+                        .tick =
+                            if zero_for_one {
+                                step_tick_next - IntegerTrait::<i32>::new(1, false)
+                            } else {
+                                step_tick_next
+                            };
+                } else if state.sqrt_price_X96 != step_sqrt_price_start_X96 {
+                    // recompute unless we're on a lower tick boundary (i.e. already transitioned ticks), and haven't moved
+                    state.tick = TickMath::get_tick_at_sqrt_ratio(state.sqrt_price_X96);
+                };
+            };
+
+            // update tick and write an oracle entry if the tick change
+            let mut slot_0 = self.slot_0.read();
+            if state.tick != slot_0_start.tick {
+                slot_0.tick = state.tick;
+            }
+            slot_0.sqrt_price_X96 = state.sqrt_price_X96;
+            self.slot_0.write(slot_0);
```

Comment:
Could there be an error? The assignment of the `slot_0.sqrt_price_X96` is done within an "else" block.
 ```
// update tick and write an oracle entry if the tick change
        if (state.tick != slot0Start.tick) {
            (uint16 observationIndex, uint16 observationCardinality) =
                observations.write(
                    slot0Start.observationIndex,
                    cache.blockTimestamp,
                    slot0Start.tick,
                    cache.liquidityStart,
                    slot0Start.observationCardinality,
                    slot0Start.observationCardinalityNext
                );
            (slot0.sqrtPriceX96, slot0.tick, slot0.observationIndex, slot0.observationCardinality) = (
                state.sqrtPriceX96,
                state.tick,
                observationIndex,
                observationCardinality
            );
        } else {
            // otherwise just update the price
            slot0.sqrtPriceX96 = state.sqrtPriceX96;
        }
```
```suggestion
            if state.tick != slot_0_start.tick {
                slot_0.tick = state.tick;
            } else {
                 slot_0.sqrt_price_X96 = state.sqrt_price_X96;
            }
```

Reply:
In the `if` statement, the assignment of `slot_0.sqrt_price_X96` is also performed. 
```
// update tick and write an oracle entry if the tick change
if (state.tick != slot0Start.tick) {
   ...
   (slot0.sqrtPriceX96, ...) = (state.sqrtPriceX96, ...);
} else {
   slot0.sqrtPriceX96 = state.sqrtPriceX96;
}
```
The code is equivalent